agents/appmotion_motor/sat1_motors_agent.py

```diff
@@ -0,0 +1,717 @@
+                return False, "Could not acquire lock"                                
+            self.motors.blockWhileMoving(motor, updatePeriod, verbose)
+
+        return True, "Motor {} stopped moving".format(motor)
+
+    def killAllCommands(self, session, params=None):
+        """
+        Stops all active commands on the device. Does not interact with lock file...
+        params: {'motor': int}
+        motor: 1,2,3. Determines which motor, 3 is for all motors. (default 3)
+        """
+        
+        motor = params.get('motor', 3)
+        with self.lock.acquire_timeout(1, job=f"killAllCommands_motor{motor}") as acquired:
+            if not acquired:
+                self.log.warn(f'Could not killAllCommands because lock held by {self.lock.job}')
+                return False, "Could not acquire lock"                                
+            self.motors.killAllCommands(motor)
+    
+        return True, "Killing all active commands on motor {}".format(motor)
+    
+    def setEncoderValue(self, session, params=None):
+        """
+        Set the encoder values in order to keep track of absolute position
+        Parameters:
+            params: {'motor': int, 'value': float}
+            motor: 1,2,3. Determines which motor, 3 is for all motors. (default 1)
+            value: Sets encoder value. (default 0)
+        """
+
+        motor = params.get('motor', 1)
+        value = params.get('value', 0)
+        self.move_status = self.motors.isMoving(motor)
+        if self.move_status:
+            return False, "Motors are already moving."
+        
+        with self.lock.acquire_timeout(1, job=f"setEncoderValue_motor{motor}") as acquired:
+            if not acquired:
+                self.log.warn(f'Could not setEncoderValue because lock held by {self.lock.job}')
+                return False, 'Could not acquire lock'                                    
+            ePositions = self.motors.setEncoderValue(motor, value)
+
+        return True, "Setting encoder position to {}".format(ePositions)
+
+    def getEncoderValue(self, session, params = None):
+        """
+        Retrieve all motor step counts to verify movement.
+        Parameters:
+            params: {'motor': int}
+            motor: 1,2,3. Determines which motor, 3 is for all motors. (default 3)
+        """
+        
+        motor = params.get('motor', 3)
+        with self.lock.acquire_timeout(1, job=f"getEncoderInfo_motor{motor}") as acquired:
+            if not acquired:
+                self.log.warn(f'Could not getEncoderInfo because lock held by {self.lock.job}')
+                return False, 'Could not acquire lock'                                    
+            ePositions = self.motors.retrieveEncoderInfo(motor)
+
+        return True, ("Current encoder positions: {}".format(ePositions),ePositions)    
+    
+    def getPositions(self, session, params = None):
```

Comment:
- `getPositions` references the `Motor.propDict` "quasi attributes". These attributes only get updated upon move_position tasks. So can think of these as discretely updated values. Can return position in inches or in encoder values.
- `retrieveEncoderInfo` queries the motor controllers for the current encoder position. Can think of these values as 'continuously' updated values. Returns only encoder values.

Reply:
`get_position_in_inches` is being published to the feed. `get_position` is not being published because `retrieve_encoder_info` is already publishing the position data in counts. A little redundant, so maybe something to fix

Reply:
Alright, I guess I'm still wondering why the separate task for checking positions when they're grabbed during the acq process. Maybe they're checked via this task before doing something else? In which case it'd be useful to store the resulting positions in the `session.data` object, as that allows programmatic handling of the returned positions, rather than only showing them in the returned message that says `f"Current motor positions: {positions}"`.

Reply:
Hi Brian, we actually ended up using the different tasks as a method of debugging and then for analysis. 
![Screenshot from 2022-05-31 08-32-46](https://user-images.githubusercontent.com/56327599/171212326-2cd22f46-a4d3-42cd-a100-db78fa4571ff.png)
(attached a screenshot of the grafana panel i used during bmap analysis in case it helps!)

- `get_position_in_inches` was used to see if the motor received a command to move and where it was told to move to (in inches so I could understand where it was going). This is due to how the position value is set, basically as input decided from the user.
- `retrieve_encoder_info` was used primarily in analysis. I used the _encoder values_ to create beam maps. These values are reported from the motor controller, so I use these as the 'actual position' of the motors. 

But yeah, we'll put the resulting positions in a `session.data` object!

Reply:
Positions in inches and positions in encoder counts are now included in `session.data`.